### ISSUE TITLE
Remove deprecated support for aria-invalid state on the Radio Component

### ISSUE THREAD

**Issue by jnjosh** [adobe/spectrum-web-components#3571]
### Code of conduct

- [X] I agree to follow this project's code of conduct.

### Description of issue

In [updating documentation](https://github.com/adobe/spectrum-web-components/pull/3558) on properly showing invalid state in radio groups (#3005). We discussed removing the `invalid` attribute from the Radio component—favoring the Radio Group instead. In talking with the accessibility team we learned:

> In WAI-ARIA 1.2, the aria-invalid state is deprecated on role="radio" (see “inherited states and properties” under https://www.w3.org/TR/wai-aria-1.2/#radio), and I don’t think that screen readers will be consistent about announcing the invalid state when aria-invalid is specified on a input[type=radio], adding aria-invalid to the role="radiogroup" is the recommended approach, (see “supported states and properties” under https://www.w3.org/TR/wai-aria-1.2/#radiogroup).

The Radio component itself will apply this deprecated `invalid` attribute in `updated(changes:)` which based on this new understanding, is deprecated behavior. The component should be updated to properly represent invalid state in the Radio Group, not on the individual Radio element.

Discussion from PR: https://github.com/adobe/spectrum-web-components/pull/3558#discussion_r1301557678

**Comment by najikahalsema** [comment]
We should check to see if there are any other deprecated ARIA attributes, in radio or otherwise.

**Comment by castastrophe** [comment]
Since we are tracking this information internally, I'm going to close out this issue. Please let me know if you need anything else from this ticket though!
